src/v2/Apps/Partner/Components/PartnerHeader/PartnerHeaderImage.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { PartnerHeaderImage_profile } from "v2/__generated__/PartnerHeaderImage_profile.graphql"
+import { FullBleed } from "v2/Components/FullBleed"
+import styled from "styled-components"
+
+const Container = styled(FullBleed)`
+  position: relative;
+  overflow: hidden;
+  background-attachment: fixed;
+`
+
+const Image = styled.img<{ scrollPosition: number }>`
+  display: block;
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+  transform: translate(0px, ${props => props.scrollPosition}px);
+`
+
+interface PartnerHeaderImageProps {
+  profile: PartnerHeaderImage_profile
+}
+
+export const PartnerHeaderImage: React.FC<PartnerHeaderImageProps> = ({
+  profile,
+}) => {
+  const [scrollPosition, setScrollPosition] = useState(0)
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll)
+    return () => {
+      window.removeEventListener("scroll", handleScroll)
+    }
+  }, [])
```

Comment:
Could you add a screen capture of what this looks like? Would probably want to [make the listener passive](https://web.dev/uses-passive-event-listeners/)

Reply:
![1](https://user-images.githubusercontent.com/79979820/112173513-b0de8e80-8c06-11eb-91f5-dc009f009823.gif)

Reply:
It was not recorded, but sometimes it flashes and twitches.

Reply:
Maybe it makes sense to rewrite it with [background-attachment](https://developer.mozilla.org/en-US/docs/Web/CSS/background-attachment)

Reply:
Does adding a Z dimension of 0 on the translate fix the flashing?

Reply:
No, I tried `transform: translate(0px, 0px) translateZ(0px);` and `transform: translate3d(0px, 0px, 0px)` but result looks the same. The worst result that I had:
![2](https://user-images.githubusercontent.com/79979820/112179490-cb673680-8c0b-11eb-8c45-dd1019aca4bb.gif)

It happened when I reload the page not from the initial page position and started scrolling.

Reply:
Maybe the problem with my pc? I will try to run it at home.

Reply:
If you want to merge a version of this and I can take a look locally that's fine.

`background-attachment` is a no go unfortunately: we'd lose srcSet + adaptive resolution + lazy loading, which make a huge difference on metrics. And performance should be a priority over this feature.

But I suspect we can figure something out.

Reply:
@dzucconi  So, I just added new version that use requestAnimationFrame loop, I think it should have the best performance. Unfortunately, I steal receive some flashing when page initialising. Could you please take a look?

Reply:
OK so yeah it looks like it's just caused by the optimizations here. The way I see it these are the options:

* Pure CSS solution (we do this on the homepage FWIW though it's not a great example)

```tsx
  return (
    <>
      {/* Spacer to prevent space from collapsing*/}
      <Box height={[280, 600]} />

      <Container
        position="fixed"
        top={[MOBILE_NAV_HEIGHT, NAV_BAR_HEIGHT]}
        height={[280, 600]}
        bg="black10"
      >
        <picture>
          <source srcSet={image.lg.srcSet} media="(min-width: 1200px)" />
          <source srcSet={image.md.srcSet} media="(min-width: 700px)" />
          <source srcSet={image.sm.srcSet} media="(max-width: 700px)" />
          <Image src={image.sm.src} alt="" loading="lazy" />
        </picture>
      </Container>
    </>
```

Then restructure the header and app components a bit such that the remainder is a foreground which scrolls over it.

Alternatively if we want to keep the JS solution:

* Add a short transition on the transform property so at least the positioning lag is smooth

Reply:
Thanks! I'll check both and get back to you tomorrow.

Reply:
@dzucconi I used pure css variant and it works much better. Thanks for the help!

![ezgif com-gif-maker (4)](https://user-images.githubusercontent.com/79979820/112447479-6bcf6f00-8d62-11eb-8020-71e82c155261.gif)